src/test/function_test/test_split.cpp

```diff
@@ -0,0 +1,356 @@
+            while (scanner->next(hash_key, sort_key, expected_value, &info) == 0) {
+                ASSERT_EQ(expected[hash_key][sort_key], expected_value);
+                count++;
+            }
+        }
+        ASSERT_EQ(count, dataset_count);
+        for (auto scanner : scanners) {
+            delete scanner;
+        }
+    }
+
+public:
+    std::shared_ptr<replication_ddl_client> ddl_client;
+    pegasus_client *pg_client;
+    std::unordered_map<std::string, std::unordered_map<std::string, std::string>> expected;
+    const int32_t partition_count = 4;
+    const int32_t dataset_count = 1000;
+    const std::string dataset_hashkey_prefix = "hashkey";
+    const std::string dataset_sortkey_prefix = "sortkey";
+    const std::string splitting_hashkey_prefix = "keyh_";
+    const std::string splitting_sortkey_prefix = "keys_";
+    const std::string data_value = "vaule";
+    int32_t count_during_split = 0;
+};
+
+TEST_F(partition_split_test, split_with_write)
+{
+    const std::string table_name = "split_table_1";
+    SetUp(table_name);
+    start_partition_split(table_name);
+    // write data during partition split
+    do {
+        write_data_during_split();
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+    } while (!is_split_finished(table_name));
+    std::cout << "Partition split succeed" << std::endl;
+
+    verify_data_after_split();
+    TearDown(table_name);
+}
+
+TEST_F(partition_split_test, split_with_read)
+{
+    const std::string table_name = "split_table_2";
+    SetUp(table_name);
+    start_partition_split(table_name);
+    // read data during partition split
+    do {
+        read_data_during_split();
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+    } while (!is_split_finished(table_name));
+    std::cout << "Partition split succeed" << std::endl;
+    verify_data_after_split();
+    TearDown(table_name);
+}
+
+TEST_F(partition_split_test, split_with_scan)
+{
+    const std::string table_name = "split_table_3";
+    SetUp(table_name);
+    start_partition_split(table_name);
+    int32_t count = 0;
+    do {
+        hash_scan_during_split(count);
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+        ++count;
+    } while (!is_split_finished(table_name));
+    std::cout << "Partition split succeed" << std::endl;
+    verify_data_after_split();
+    std::this_thread::sleep_for(std::chrono::seconds(30));
+    full_scan_after_split();
+    TearDown(table_name);
+}
+
+TEST_F(partition_split_test, pause_split)
+{
+    const std::string table_name = "split_table_4";
+    SetUp(table_name);
+    start_partition_split(table_name);
+
+    bool already_pause = false, already_restart = false;
+    int32_t target_partition = 2, count = 30;
+    do {
+        write_data_during_split();
+        // pause target partition split
+        if (!already_pause &&
+            check_partition_split_status(table_name, -1, split_status::SPLITTING)) {
```

Comment:
`-1` makes me confused, can you give it a name?

Reply:
-1 means it will check all partitions, you can reference `check_partition_split_status` and `is_single_partition`